packages/@aws-cdk/aws-synthetics/lib/code.ts

```diff
@@ -127,18 +128,25 @@ export class AssetCode extends Code {
    *
    * @param handler the canary handler
    */
-  private validateCanaryAsset(handler: string, family: RuntimeFamily) {
+  private validateCanaryAsset(scope: Construct, handler: string, family: RuntimeFamily) {
```

Comment:
I suggest you add this at the beginning of this function:

```ts
if (!this.asset) {
    throw new Error("'validateCanaryAsset' must be called after 'this.asset' is defined");
}

Reply:
Seems reasonable and a better gaurd!